### PR TITLE
Add lexer for graphviz

### LINKED ISSUE #731: new lexer request: dot (graphviz)
(Original issue 1024 created by psuter on 2014-08-12T08:29:53.471324+00:00)

The DOT language is used by the open source graph visualization software Graphviz to represent structural information as diagrams of abstract graphs and networks.

The language grammar is described at http://www.graphviz.org/doc/info/lang.html

An example:

```
#!dot
    digraph G {
       Hello->Pygments
    }
```

Would be nice to have a lexer for this in Pygments.

### PR DESCRIPTION
**Pull request by nikeee**
Discussion here: #731

Resolves #731

### CODE REVIEW

**Comment by Anteru** [comment]
Thanks for the contribution! Can you please add an example file as well?

**Comment by nikeee** [comment]
I'll take som inspiration from this PR: https://github.com/pygments/pygments/pull/1519/files
Seems that there is something missing in other files as well.

**Comment by prodigion** [comment]
There's some examples here that you could copy: https://github.com/xflr6/graphviz/tree/master/examples/

**Comment by nikeee** [comment]
I updated the PR.

File: doc/languages.rst [hunk 1]
```diff
@@ -51,6 +51,7 @@ Programming languages
 * `Dart <https://dart.dev/>`_
 * DCPU-16
 * `Delphi <https://www.embarcadero.com/products/delphi>`_
+* `DOT / Graphviz <https://graphviz.org>`
```
Comment (Anteru) [inline doc/languages.rst:54]: Missing trailing `_`.

File: pygments/lexers/graphviz.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+"""
+    pygments.lexers.graphviz
+    ~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Lexer for the DOT language (graphviz).
+
+    :copyright: Copyright 2006-2020 by the Pygments team, see AUTHORS.
```
Comment (Anteru) [inline pygments/lexers/graphviz.py:8]: 2021 by now, sorry it took us so long.

File: pygments/lexers/graphviz.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+
+__all__ = ['GraphvizLexer']
+
+
+class GraphvizLexer(RegexLexer):
+    """
+    For graphviz DOT graph description language.
+
+    .. versionadded:: 2.3.0
```
Comment (Anteru) [inline pygments/lexers/graphviz.py:23]: Please bump to 2.8

**Review by Anteru — CHANGES_REQUESTED** [review]
Thanks for the contribution, only very minor changes needed before we can land this.

File: doc/languages.rst (hunk 1 above)
Comment (nikeee) [reply doc/languages.rst]: Done

File: pygments/lexers/graphviz.py (hunk 2 above)
Comment (nikeee) [reply pygments/lexers/graphviz.py]: Well, I didn't expect it being merged in 2020. We have time for that!

File: pygments/lexers/graphviz.py (hunk 3 above)
Comment (nikeee) [reply pygments/lexers/graphviz.py]: Done

**Comment by nikeee** [comment]
I pushed an update ot address the stuff that came up in the review.

**Review by Anteru — APPROVED** [review]

**Comment by Anteru** [comment]
Merged for the next release, thanks for your contribution!

**Pull request merged**